vision/src/main/java/coppercore/vision/CameraIOPhoton.java

```diff
@@ -90,33 +87,29 @@ public void updateInputs(CameraIOInputs inputs) {
                 });
     }
 
-    private static boolean filterPhotonPose(EstimatedRobotPose photonPose) {
-        // TODO: Actual pose rejection
-        if (photonPose.targetsUsed.size() < 2) {
-            return false;
-        }
-
-        Pose3d pose = photonPose.estimatedPose;
-        // check that the pose isn't insane
-        if (pose.getZ() > 1 || pose.getZ() < -0.1) {
-            return false;
-        }
+    // NOTE: Can be used in 2025 code just not ready yet
+    // private Optional<EstimatedRobotPose> getEstimatedPose () {
+    //     Optional<EstimatedRobotPose> visionEstimate = Optional.empty();
 
-        if (calculateAverageTagDistance(photonPose)
-                > CoreVisionConstants.maxAcceptedDistanceMeters) {
-            return false;
-        }
+    //     for (var change : camera.getAllUnreadResults()) {
+    //         visionEstimate = poseEstimator.update(change);
+    //     }
 
-        return true;
-    }
+    //     return visionEstimate;
+    // }
 
-    private static double calculateAverageTagDistance(EstimatedRobotPose pose) {
+    private double calculateAverageTagDistance(EstimatedRobotPose pose) {
         double distance = 0.0;
         for (PhotonTrackedTarget target : pose.targetsUsed) {
+            var tagPose = poseEstimator.getFieldTags().getTagPose(target.getFiducialId());
+            if (tagPose.isEmpty()) {
+                continue;
+            }
```

Comment:
is it possible for `tagPose` to be empty, but still count in the `targetsUsed` list (thus making the average wrong)?

Basically: should we maintain our own count instead of using `targetsUsed.size()` below?

Reply:
yes ill update that, good catch